src/Microsoft.ApplicationInsights/DataContracts/DependencyTelemetry.cs

```diff
@@ -281,6 +305,29 @@ public override ITelemetry DeepClone()
             return new DependencyTelemetry(this);
         }
 
+        /// <summary>
+        /// In specific collectors, objects are added to the dependency telemetry which may be useful
+        /// to enhance DependencyTelemetry telemetry by <see cref="ITelemetryInitializer" /> implementations.
+        /// </summary>
+        /// <param name="key">The key of the value to get.</param>
+        /// <param name="detail">When this method returns, contains the object that has the specified key, or the default value of the type if the operation failed.</param>
+        /// <returns>true if the key was found; otherwise, false.</returns>
+        public bool TryGetOperationDetail(string key, out object detail)
+        {
+            return this.OperationDetails.TryGetValue(key, out detail);
```

Comment:
Short-circuit if operationDetails (the backing field) has not been initialized (avoid creating an empty ConcurrentDictionary here).